File: labs_python/laba3.py

import math
x = float(input('Введите значение х: '))
a = float(input('Введите значение а: '))
l = int(input("Введите сколько считать:"))
s = float(input("Введите шаг:"))
w = int(input('1 - вычислить функцию G, 2 - вычислить функцию F, 3 - вычислить функцию Y: '))
if w == 1:
    for i in range(l):
        if (4*(-4*a**2 + a*x + 5*x**2)) / (-20*a**2 + 28*a*x + 3*x**2) ==0:
            print('ERROR')
        else:
            G = (4*(-4*a**2 + a*x + 5*x**2)) / (-20*a**2 + 28*a*x + 3*x**2)
            print('G = ' + str(G))
            a += s
elif w == 2:
    for i in range(l):
        if (3,14*(27*a**2 - 105*a*x + 50*x**2)) / (3,14*(27*a**2 - 105*a*x + 50*x**2) and (3,14*(27*a**2 - 105*a*x + 50*x**2)) / (3,14*(27*a**2 - 105*a*x + 50*x**2) > 1:
            print('ERROR')
        else:
            F == sin(3,14*(27*a**2 - 105*a*x + 50*x**2)) / (3,14*(27*a**2 - 105*a*x + 50*x**2)
            print('F = ' + str(F))
            a += s
elif w == 3:
    for i in range(l):
        if (-2*a**2 - a*x + x**2 + 1):
            print('ERROR')
        else:
            Y = math.log(-2*a**2 - a*x + x**2 + 1)
            print('Y = ' + str(Y))
            a += s
else:
    print('ERROR')

File: labs_c/laba3.c
#include <stdio.h>
#include <math.h>
#include <locale.h>
int main (void)
{
setlocale(LC_ALL, "Rus");
float a, x, g, f, y, s;
int word, l, i = 0;

printf("Введите числа, не равные нулю:\n");
printf("Введите х: ");
scanf("%f", &x); printf("Введите a: "); scanf("%f", &a);

printf("Введите сколько считать: ");
scanf("%i",&l);

printf("Введите шаг: ");
scanf("%f",&s);

printf("1 - вычислить функцию G, 2 - вычислить функцию F, 3 - вычислить функцию Y:");
scanf("%i",&word);

for (i = 1; i <= l; i++)
    switch(word)
    {
        case 1:
            if ((-20*a*a + 28*a*x + 3*x*x) == 0)
            {
                printf("ERROR");
                x += s;
                a += s;
            }
            else
            {
                g = (4*(-4*a*a + a*x + 5*x*x)) / (-20*a*a + 28*a*x + 3*x*x);
                printf("g=%f\n\n",g);
                x += s;
                a += s;
                break;
            }
        case 2:
            f = sin(3,14*(27*a*a - 105*a*x + 50*x*x)) / (3,14*(27*a*a - 105*a*x + 50*x*x);
            printf("f=%f\n\n",f);
            x += s;
            a += s;
            break;
        case 3:
            if ((-2*a*a - a*x + x**2 + 1) <= 0)
            {
                printf("ERROR");
                x += s;
                a += s;
            }
            else
            {
                y = math.log(-2*a*a - a*x + x**2 + 1);
                printf("y=%f\n\n",y);
                x += s;
                a += s;
                break;
            }
        default:
            printf(" неправильный ввод.\n");
}
}
